packages/cozy-procedures/src/redux/documentsDataSlice.js

```diff
@@ -0,0 +1,116 @@
+import { createSlice } from 'redux-starter-kit'
+import get from 'lodash/get'
+import creditApplicationTemplate from '../templates/creditApplicationTemplate'
+import { AdministrativeProcedure } from 'cozy-doctypes'
+
+const documentsSlice = createSlice({
+  initialState: {
+    data: {
+      /*  payslip: {
+        files: []
+      } */
+    },
+    ui: {
+      /* payslip: {
+        loading: true,
+        error: false
+      } */
+      initiated: false
+    }
+  },
+  slice: 'documents',
+  reducers: {
+    init: (state, action) => {
+      return {
+        data: Object.keys(action.payload).reduce((acc, fieldId) => {
+          acc[fieldId] = {
+            files: []
+          }
+          return acc
+        }, {}),
+        ui: Object.keys(action.payload).reduce((acc, fieldId) => {
+          acc[fieldId] = {
+            loading: false,
+            error: false
+          }
+          return acc
+        }, {})
+      }
+    },
+
+    fetchDocumentLoading: (state, action) => {
+      const { idDoctemplate, loading } = action.payload
+      state.ui[idDoctemplate].loading = loading
+    },
+    fetchDocumentSuccess: (state, action) => {
+      const { idDoctemplate, loading, files } = action.payload
+      state.ui[idDoctemplate].loading = loading
```

Comment:
nit: I prefer to set `loading` to `false` directly here. I can't find an example where we would want to dispatch success and `loading` to `true`